torch/csrc/distributed/rpc/tensorpipe_utils.cpp

```diff
@@ -91,11 +91,54 @@ class TensorpipeCpuConverter : public TensorpipeDeviceTypeConverter {
 
 C10_REGISTER_TENSORPIPE_DEVICE_TYPE_CONVERTER(CPU, TensorpipeCpuConverter);
 
+struct MetaBuffer {
+  void* ptr{nullptr};
+
+  tensorpipe::Device getDevice() const {
+    return tensorpipe::Device{tensorpipe::kMetaDeviceType, 0};
+  }
+};
+
+class TensorpipeMetaConverter : public TensorpipeDeviceTypeConverter {
+ public:
+  c10::optional<std::vector<char>> prepareTensorForSending(
+      const c10::Storage& storage,
+      const std::vector<c10::Stream>& /* streams */,
+      tensorpipe::Message& message) const override {
+    MetaBuffer buffer;
+    tensorpipe::Message::Tensor tensor;
+    tensor.buffer = buffer;
```

Comment:
If the buffer length is zero, do you need to assign it at all? (I'm also not terribly familiar with tensorpipe::Message::Tensor, so I'm not sure how you could be assigning a buffer that TensorPipe ostensibly knows nothing about to it)

Reply:
agree, it is reverted